pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -887,6 +898,95 @@ func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMe
 	return err
```

Comment:
We should return a requeue here instead of nil, because the Machine itself has been updated so further attempts to write to it will fail.